packages/react/index.tsx

```diff
@@ -1 +1,6 @@
+import { I18n } from '@aws-amplify/core';
+import { dict } from '../core';
+
 export * from './src';
+
+I18n.putVocabularies(dict);
```

Comment:
We'll need to find another place for this that's not a side-effect.  Unfortunately, that may mean Vue's `setup()` function, React's `Authenticator` with `useEffect`, etc.

I personally had to deal with Angular 9's tree-shaking behavior with side-effects like these that were getting stripped in production builds (but there in development).

Amplify JS lost some bundle optimizations as a result:

- https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free
- https://github.com/aws-amplify/amplify-js/pull/6867
- https://github.com/aws-amplify/amplify-js/pull/6844

Another thing we can do is, rather than `I18n.putVocabularies(dict)` by default, we can instruct customers in our docs to do something like:

> https://quip-amazon.com/UtYxAfbJ7zwz/Intern-Project-Report-Amplify-Issue-Librarian

```js
import de from "@aws-amplify/ui-react/locales/de/translations.json"

I18n.putVocabularies({ de })
```

or

```js
import { de } from "@aws-amplify/ui-react/translations"

I18n.putVocabularies({ de })
```

I'm not sure which is better (the latter seems "cleaner" to me)

Reply:
Oh that's a good idea. Leave it up to the customers. I like that more. Will remove the `.putVocabularies` for now.